jax/experimental/sparse/bcsr.py

```diff
@@ -231,17 +251,45 @@ def _bcsr_fromdense_abstract_eval(mat, *, nse, n_batch, n_dense, index_dtype):
 def _bcsr_fromdense_batching_rule(batched_args, batch_dims, *, nse, n_batch,
                                   n_dense, index_dtype):
   M, = batched_args
-  if batch_dims != (0,):
-    raise NotImplementedError(f"{batch_dims=}")
-  new_n_batch = n_batch + 1
-  n_sparse = M.ndim - n_dense - new_n_batch
-  if n_sparse != 2:
-    raise ValueError("_bcsr_fromdense_batching_rule: must have 2 sparse "
-                     f"dimensions but {n_sparse} is given.")
-  return _bcsr_fromdense(M, nse=nse, n_batch=new_n_batch, n_dense=n_dense,
-                         index_dtype=index_dtype), (0, 0, 0)
+  bdim, = batch_dims
```

Comment:
User error can't cause this to fail right? Like you could assert it's len == 1 ?

Reply:
I think user input is validated at a higher level in vmap, so at this point it's safe to assume the arguments have the expected shape.